src/TypeGen/TypeGen.Core/Generator.cs

```diff
@@ -66,11 +66,79 @@ public Generator()
                 new TsContentParser(_fileSystem));
         }
 
+        /// <summary>
+        /// Generates an `index.ts` file which exports all types within the generated files
+        /// </summary>
+        /// <param name="generatedFiles"></param>
+        public string GenerateIndexFile(IEnumerable<string> generatedFiles)
+        {
+            string typeScriptFileExtension = "";
+            if (!string.IsNullOrEmpty(Options.TypeScriptFileExtension))
+            {
+                typeScriptFileExtension = "." + Options.TypeScriptFileExtension;
+            }
+
+            string exports = generatedFiles.Aggregate("", (prevExports, file) =>
+            {
+                string fileNameWithoutExt = file.Remove(file.Length - typeScriptFileExtension.Length).Replace("\\", "/");
+                return prevExports + _templateService.FillIndexExportTemplate(fileNameWithoutExt);
+            });
+            string content = _templateService.FillIndexTemplate(exports);
+
+            string filename = "index" + typeScriptFileExtension;
+            _fileSystem.SaveFile(Path.Combine(Options.BaseOutputDirectory, filename), content);
+            return filename;
+        }
+
+        /// <summary>
+        /// Generates TypeScript files for C# files in assemblies
+        /// </summary>
+        /// <param name="assemblies"></param>
+        public GenerationResult Generate(IEnumerable<Assembly> assemblies)
+        {
+            return Generate(assemblies, false);
+        }
+
+        /// <summary>
+        /// Generates TypeScript files for C# files in assemblies
+        /// </summary>
+        /// <param name="assemblies"></param>
+        /// <param name="skipCreateIndexFile">Wheter to ignore Options.CreateIndexFile</param>
+        public GenerationResult Generate(IEnumerable<Assembly> assemblies, bool skipCreateIndexFile)
```

Comment:
Can you give an example when you would use `skipCreateIndexFile` instead of setting `Options.CreateIndexFile`?

Reply:
The problem was that calling `Generate(IEnumerable<Assembly> assemblies)` should generate an index file. That function calls `GenerationResult Generate(Assembly assembly)`, which on itself should also create an index file, but we don't want to temporary recreate the index file for each assembly, if it nevertheless gets written at the end. But I just cleaned up that mess, it seems like yesterday I was too stressed.

Reply:
No problem, it's fine! :) I wanted to know what happens in the code essentially.